src/corner-dialog/src/CornerDialog.js

```diff
@@ -222,28 +67,75 @@ export default class CornerDialog extends PureComponent {
       cancelLabel,
       confirmLabel,
       onOpenComplete,
+      onCloseComplete,
+      onCancel,
+      onConfirm,
       containerProps = {},
       position
-    } = this.props
+    } = props
+
+    const [exiting, setExiting] = useState(false)
+    const [exited, setExited] = useState(!props.isShown)
+
+    useEffect(() => {
+      if (isShown) {
+        setExited(false)
+      }
+    }, [isShown])
+
+    const handleExited = useCallback(() => {
+      setExiting(false)
+      setExited(true)
+
+      onCloseComplete()
+    }, [onCloseComplete])
+
+    const handleClose = useCallback(() => setExiting(true))
 
-    const { exiting, exited } = this.state
+    const handleCancel = useCallback(() => {
+      onConfirm(handleClose)
+    }, [onCancel])
+
+    const handleConfirm = useCallback(() => {
+      onCancel(handleClose)
+    }, [onConfirm])
+
+    const renderChildren = useCallback(() => {
+      if (typeof children === 'function') {
+        return children({ close: handleClose })
+      }
+
+      if (typeof children === 'string') {
+        return (
+          <Paragraph size={400} color="muted">
+            {children}
+          </Paragraph>
+        )
+      }
+
+      return children
+    }, [children])
+
+    if (exited) {
+      return null
+    }
 
-    if (exited) return null
     return (
       <Portal>
         <Transition
           appear
           unmountOnExit
           timeout={ANIMATION_DURATION}
           in={isShown && !exiting}
-          onExited={this.handleExited}
+          onExited={handleExited}
           onEntered={onOpenComplete}
         >
           {state => (
             <Card
               role="dialog"
               backgroundColor="white"
               elevation={4}
+              ref={ref}
```

Comment:
Actually not sure we need a ref here... What would you be attempting to do with a ref for a dialog?

Reply:
Good q! I think these actually could do without `forwardRef` support, since we're not spreading props to the underlying element. Will update.